File: quasar/Quasar.DataTransfer/Vendas/Responses/ItemVendaResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Quasar.DataTransfer.Produtos.Responses;

namespace Quasar.DataTransfer.Vendas.Responses
{
    public class ItemVendaResponse
    {
        public int Codigo {get; set;}
        public int Quantidade {get; set;}
        public ProdutoResponse Produto {get; set;}
        public decimal ValorUnitario {get; set;}
    }
}

File: quasar-front/src/app/shared/models/responses/itemVenda.response.ts
import { ProdutoResponse } from 'src/app/shared/models/responses/produto.response';

export class ItemVendaResponse {
  public codigo: number;
  public quantidade: number;
  public produto: ProdutoResponse;
  public valorUnitario: number;

  constructor(params: Partial<ItemVendaResponse>) {
    this.codigo = params.codigo || 0;
    this.quantidade = params.quantidade || 0;
    this.produto = params.produto || new ProdutoResponse({});
    this.valorUnitario = params.valorUnitario || 0;
  }
}

